lib/vtls/vtls.c

```diff
@@ -1239,31 +1239,32 @@ static size_t Curl_multissl_version(char *buffer, size_t size)
 
   if(current != selected) {
     char *p = backends;
+    char *end = backends + sizeof(backends);
     int i;
 
     selected = current;
 
-    for(i = 0; available_backends[i]; i++) {
+    for(i = 0; available_backends[i] && p < (end - 4); i++) {
       if(i)
         *(p++) = ' ';
       if(selected != available_backends[i])
         *(p++) = '(';
-      p += available_backends[i]->version(p, backends + sizeof(backends) - p);
+      p += available_backends[i]->version(p, end - p - 2);
```

Comment:
Shouldn't this be an `end - p > 2 ? end - p - 2 : 0`? The second parameter is declared as `size_t`, i.e. unsigned...

Reply:
end - p is always > 2 at that point, refer to the loop condition `p < (end - 4)`, then after that `p` can be incremented twice before `end - p - 2`

Reply:
Right, unless I'm missing something @jay is right, the current coding should be safe (but it might warrant a comment explaining it).

Reply:
I find adding a comment to be unnecessary I think it's pretty straightforward.

Reply:
Well, one could just right the loop condition as `...&& (end - p) > 4` and it'd be easier to read

Reply:
> I find adding a comment to be unnecessary I think it's pretty straightforward.

So you're saying that I am dumb. Because I did not see that.

I don't take this personally, coming from you, as I had pleasant and productive interactions with you in the past, but I wonder whether you realized how unfortunate a reaction your comment would elicit (and I think it is pretty obvious that it would do exactly that). Might want to be more careful.

In short: while it might be obvious to you, *now*, I think that my comment is testament enough to give evidence to the contrary. And who knows, maybe you yourself will belong into the same "wait, what?" camp in 6 months from now.

Reply:
I'm in @dscho's corner here. When I first saw the question here about this line of code it was not immediately obvious to me that it was right. A comment could've remedied that both for @dscho and for me. For those who don't need the comment, it won't hurt (much) anyway.

Reply:
> > I find adding a comment to be unnecessary I think it's pretty straightforward.
> 
> So you're saying that I am dumb. Because I did not see that.
> 
> I don't take this personally, coming from you, as I had pleasant and productive interactions with you in the past, but I wonder whether you realized how unfortunate a reaction your comment would elicit (and I think it is pretty obvious that it would do exactly that). Might want to be more careful.
>
> In short: while it might be obvious to you, _now_, I think that my comment is testament enough to give evidence to the contrary. And who knows, maybe you yourself will belong into the same "wait, what?" camp in 6 months from now.

Whoa I may not be very tactful but don't put words in my mouth. I figured you misread the code, like I do occasionally. I would've shared my opinion even if that wasn't the case. If you had a different opinion that's all you needed to say in your reply.

Anyway, clearly my opinion of it being straightforward is not in favor. Someone propose a comment so we can put this to bed. `/* end - p is > 2 at this point */` ?